paddle/fluid/framework/ir/memory_optimize_pass/memory_optimization_var_info.h

```diff
@@ -44,17 +48,30 @@ class MemOptVarInfo {
     runtime_ref_cnt_ = ref_cnt;
   }
 
-  bool IsSkipped() const { return skipped_; }
+  // Skip all memory optimization, including memory reuse and garbage collection
+  void SetSkipAllMemoryOptimization(bool is_skipped) {
+    skip_all_memory_optimization_ = is_skipped;
+  }
 
-  void SetSkip(bool skipped) { skipped_ = skipped; }
+  bool IsSkippedAllMemoryOptimization() const {
+    return skip_all_memory_optimization_;
+  }
+
+  // Skip all memory reuse, including inplace and cross op memory reuse
+  void SetSkipMemoryReuse(bool is_skipped) { skip_memory_reuse_ = is_skipped; }
+
+  bool IsSkippedMemoryReuse() const {
+    return skip_memory_reuse_ || skip_all_memory_optimization_;
+  }
 
   const std::string &Name() const { return name_; }
 
  private:
   std::string name_;
   size_t ref_cnt_;
```

Comment:
Please add comment for the difference of these two variables.

Reply:
I meant, ref_cnt_ and runtime_ref_cnt_

Reply:
Done.

Reply:
Great!